### PR TITLE
Change ConfiguredTargetQuery to use KeyedConfiguredTarget as a value.

### LINKED ISSUE #11993: cquery produces non-actionable warning related to toolchains
### Description of the problem / feature request:

IF
1. there is a target Foo which depends on some toolchain;
2. `--incompatible_override_toolchain_transition` is in effect;
3 command is `bazel cquery deps(Foo)
THEN
`WARNING: Targets were missing from graph` is generated.

The small repro with Python toolchain is published [here](https://github.com/Bazel-snippets/cquery_targets_missing_repro)
We get exactly the same problem with C++ toolchain as well.

### What operating system are you running Bazel on?

Windows, Linux, OSX.

### What's the output of `bazel info release`?

3.4.1

### PR DESCRIPTION
**Pull request by katre**
This will allow us to keep the configured target key between lookup rounds.

Fixes #11993.

### CODE REVIEW

File: src/main/java/com/google/devtools/build/lib/query2/cquery/ConfiguredTargetQueryEnvironment.java [hunk 1]
```diff
@@ -517,18 +517,15 @@ protected BuildConfiguration getConfiguration(ConfiguredTarget target) {
   }
 
   @Override
-  protected ConfiguredTargetKey getSkyKey(ConfiguredTarget target) {
-    return ConfiguredTargetKey.builder()
-        .setConfiguredTarget(target)
-        .setConfiguration(getConfiguration(target))
-        .build();
+  protected ConfiguredTargetKey getSkyKey(KeyedConfiguredTarget target) {
```
Comment (juliexxia) [inline src/main/java/com/google/devtools/build/lib/query2/cquery/ConfiguredTargetQueryEnvironment.java:520]: I think this method's assumptions have caused other issues in the past so it's nice we can tighten it up here.

File: src/test/java/com/google/devtools/build/lib/query2/cquery/ConfiguredTargetQuerySemanticsTest.java [hunk 2]
```diff
@@ -120,9 +119,9 @@ public void testLabelFunction_getCorrectlyConfiguredDeps() throws Exception {
     setUpLabelsFunctionTests();
 
     // Test that this retrieves the correctly configured version(s) of the dep(s).
-    ConfiguredTarget patchDep =
+    KeyedConfiguredTarget patchDep =
```
Comment (juliexxia) [inline src/test/java/com/google/devtools/build/lib/query2/cquery/ConfiguredTargetQuerySemanticsTest.java:122]: Can we add a test case for the situation that caused #11993 ?

**Review by juliexxia — COMMENTED** [review]
Overall looks great!

File: src/main/java/com/google/devtools/build/lib/query2/cquery/ConfiguredTargetAccessor.java [hunk 3]
```diff
@@ -180,14 +176,17 @@ public Target getTargetFromConfiguredTarget(ConfiguredTarget configuredTarget) {
   }
 
   /** Returns the rule that generates the given output file. */
-  public RuleConfiguredTarget getGeneratingConfiguredTarget(OutputFileConfiguredTarget oct)
+  RuleConfiguredTarget getGeneratingConfiguredTarget(KeyedConfiguredTarget kct)
```
[the remark this answers is not in the document]
Comment (katre) [reply src/main/java/com/google/devtools/build/lib/query2/cquery/ConfiguredTargetAccessor.java:179]: Done.

File: src/test/java/com/google/devtools/build/lib/query2/cquery/ConfiguredTargetQuerySemanticsTest.java (hunk 2 above)
Comment (katre) [reply src/test/java/com/google/devtools/build/lib/query2/cquery/ConfiguredTargetQuerySemanticsTest.java:122]: Good point. Added to //src/test/shell/bazel:toolchain_transition_test

File: src/test/shell/bazel/toolchain_transition_test.sh [hunk 4]
```diff
@@ -355,4 +354,29 @@ EOF
   expect_log 'extra_lib: message: extra_lib foo, target_dep: target, tool_dep: exec-alpha'
 }
 
+function test_toolchain_transition_cquery() {
```
Comment (juliexxia) [inline src/test/shell/bazel/toolchain_transition_test.sh:357]: Do you mind adding a comment about what this seemingly innocuous test case is testing for?

**Review by juliexxia — APPROVED** [review]

File: src/test/shell/bazel/toolchain_transition_test.sh (hunk 4 above)
Comment (katre) [reply src/test/shell/bazel/toolchain_transition_test.sh:360]: Done.

**Pull request closed without merging**